alerter/src/test/java/org/apache/hertzbeat/alert/service/AlertDefineYamlImExportServiceTest.java

```diff
@@ -0,0 +1,161 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import org.apache.hertzbeat.alert.dto.AlertDefineDTO;
+import org.apache.hertzbeat.alert.dto.ExportAlertDefineDTO;
+import org.apache.hertzbeat.alert.service.impl.AlertDefineYamlImExportServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * test case for {@link AlertDefineYamlImExportServiceImpl}
+ */
+
+@ExtendWith(MockitoExtension.class)
+class AlertDefineYamlImExportServiceTest {
+
+	@InjectMocks
+	private AlertDefineYamlImExportServiceImpl service;
+
+	private static final String YAML_DATA =
+			"- alertDefine:\n" +
+					"    app: App1\n" +
+					"    metric: Metric1\n" +
+					"    field: Field1\n" +
+					"    preset: true\n" +
+					"    expr: Expr1\n" +
+					"    priority: 1\n" +
+					"    times: 1\n" +
+					"    tags: []\n" +
+					"    enable: true\n" +
+					"    recoverNotice: true\n" +
+					"    template: Template1\n";
+
+	private InputStream inputStream;
+	private List<ExportAlertDefineDTO> alertDefineList;
+
+	@BeforeEach
+	public void setup() {
+
+		inputStream = new ByteArrayInputStream(YAML_DATA.getBytes(StandardCharsets.UTF_8));
+
+		AlertDefineDTO alertDefine = new AlertDefineDTO();
+		alertDefine.setApp("App1");
+		alertDefine.setMetric("Metric1");
+		alertDefine.setField("Field1");
+		alertDefine.setPreset(true);
+		alertDefine.setExpr("Expr1");
+		alertDefine.setPriority((byte) 1);
+		alertDefine.setTimes(1);
+		alertDefine.setTags(List.of());
+		alertDefine.setEnable(true);
+		alertDefine.setRecoverNotice(true);
+		alertDefine.setTemplate("Template1");
+
+		ExportAlertDefineDTO exportAlertDefine = new ExportAlertDefineDTO();
+		exportAlertDefine.setAlertDefine(alertDefine);
+
+		alertDefineList = List.of(exportAlertDefine);
+	}
+
+	@Test
+	void testParseImport() {
+
+		List<ExportAlertDefineDTO> result = service.parseImport(inputStream);
+
+		assertNotNull(result);
+		assertEquals(1, result.size());
+
+		// yaml load data to string result and alertDefineList are not equal. why?
+		// System.out.println(alertDefineList.toString());
+		// System.out.println(result.toString());
+		// assertEquals(alertDefineList, result);
```

Comment:
![image](https://github.com/user-attachments/assets/5647ff0e-b0a4-4b12-984d-0a514c2f447f)
Looks like `snakeyaml` use LinkedHashMap as its underlying data structure.
Maybe we can try to compare them in a different way.

Reply:
yep, I try to solve it.